sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/DiagnosticDescriptorBuilderTest.cs

```diff
@@ -57,5 +61,75 @@ public void GetDescriptor_SetsIsEnabledByDefaultToTrue()
                 .IsEnabledByDefault
                 .Should().BeTrue();
         }
+
+        [TestMethod]
+        public void GetDescriptor_WhenIsActivatedByDefaultAndIdeVisibilityNotHidden_HasOnlySonarWayTag()
+        {
+            // Arrange
+            var diagnosticId = "foo";
+            var mockedResourceManager = CreateMockedResourceManager(diagnosticId, true);
+
+            // Act
+            var result = DiagnosticDescriptorBuilder.GetDescriptor(diagnosticId, "", mockedResourceManager);
+
+            // Assert
+            result.CustomTags.Should().ContainSingle(DiagnosticTagsHelper.SonarWayTag);
+        }
+
+        [TestMethod]
+        public void GetDescriptor_WhenIsNotActivatedByDefaultAndIdeVisibilityNotHidden_IsEmpty()
+        {
+            // Arrange
+            var diagnosticId = "foo";
+            var mockedResourceManager = CreateMockedResourceManager(diagnosticId, false);
+
+            // Act
+            var result = DiagnosticDescriptorBuilder.GetDescriptor(diagnosticId, "", mockedResourceManager);
+
+            // Assert
+            result.CustomTags.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetDescriptor_WhenIsActivatedByDefaultAndIdeVisibilityIsHidden_HasSonarWayAndUnnecessaryTags()
+        {
+            // Arrange
+            var diagnosticId = "foo";
+            var mockedResourceManager = CreateMockedResourceManager(diagnosticId, true);
+
+            // Act
+            var result = DiagnosticDescriptorBuilder.GetDescriptor(diagnosticId, "", IdeVisibility.Hidden, mockedResourceManager);
+
+            // Assert
+            result.CustomTags.Should().ContainInOrder(DiagnosticTagsHelper.SonarWayTag, WellKnownDiagnosticTags.Unnecessary);
```

Comment:
This check is over-specific; we don't care about the order, only that those and only those tags are present.